### DISCUSSION TITLE
#27 I don't see the problem!

### DISCUSSION THREAD

**Discussion by verberg** [cssbattle/forum#208] [Q&A] [answered]
I have a new solution (target 27) and I don't see why it functions like that - there must be something width the background:

<style>*{background:radial-gradient(1q,#AA445F,106q,#E38F66)#E38F66}*>*{margin:80 130;border-radius:50%;border:32q solid;border-color:#F7EC7D#0000#F7EC7D;transform:rotate(45deg

What's with the background? If I set the background with #0000, then it gets 100% (and also with #AA445F)

when I set the border 32q solid#0000-->99.9%, 
when I set the border 32qsolid#AA445F --> 99.9%
when I set the border-color:#F7EC7D#AA445F#F7EC7D--> 99.9%

I do not understand why. My background already exists in the background-radial-gradient on *

Yes I have a solution (the radial-gradient in *>* with less chars) but I want understand why the background of *>* is somehow not transparent.
Thanks for your help!

**Comment by himanshubhatt1** [comment]
try this......




<style>*{background:radial-gradient(1q,#AA445F,106q,#E38F66)/   *remove this ...#E38F66*/   #E38F66  }*>*{margin:80 130;border-radius:50%;border:32q solid;border-color:#F7EC7D     /* for transprant use this ... #0000 */     #0000 #F7EC7D;transform:rotate(45deg

**Comment by rafehimself** [comment]
I'm not sure what do you mean
but you don't need second `#E38F66` after brackets, since `background:radial-gradient(1q,#AA445F,106q,#E38F66)` will fill whole area that is left,
`border-color:` can bet set : `#F7EC7D#00000` which means top+bottom => yellow `#F7EC7D`, left+right => transparent `#0000`
and because the size of `body` / `*>*` is smaller than 100px radius it will be entirely filled with `#AA445F` which is declared at `*` as `background:radial-gradient(1q,#AA445F,106q`

that's how I understand it, correct me if im wrong

but how do you make those cuts with just a radial-gradient 🤔 that I do not know

**Reply by verberg** [reply]
First (about the cuts): I learned from here: https://github.com/cssbattle/forum/discussions/177

And my question was  this: (sorry about my english, I 'm a German speaking woman...  ;)
A had the background set in * and then in *>* 

it sseems as if this layer (*>*) has not a background, or not this background for any reason. But if I set a the background-color also in *>* then it works.
But it works not, when I make the border width: #F7EC7D#AA445F
and olso not whenI set the default border-color with border 32qsolid#AA445F 

My actual solution is this:

<style>*>*{background:radial-gradient(1q,#AA445F,106q,#E38F66);margin:80 130;border-radius:50%;border:32q solid;border-color:#F7EC7D#0000;transform:rotate(45deg

**Reply by rafehimself** [reply]
```css
<style>*+*{--:100%0%;--p:-48vw 150q}
*{background:radial-gradient(1q at var(--,0%100%),#0000,42q,#F7EC7D 0,74q,#0000)var(--p,50vw -50vh,radial-gradient(1q,#AA445F,25vw,#E38F66
```

how is that shorter ? I don't get it 🤔

**Reply by verberg** [reply]
You have 175 chars and I have 160 chars. Just look at my solution! There I get it for you:

<style>*>*{background:radial-gradient(1q,#AA445F,106q,#E38F66);margin:80 130;border-radius:50%;border:32q solid;border-color:#F7EC7D#0000;transform:rotate(45deg

**Reply by rafehimself** [reply]
sorry, I misunderstood you, I thought you have shorter only radiant-gradient solution 🤣 😝but you just meant moving background from `*` to `*>*`

**Reply by verberg** [reply]
If you have a solution under 160 chars, just tell me!

**Reply by rafehimself** [reply]
my solution uses clip-path, but it's far away from top10

**Reply by rafehimself** [reply]
```css
<style>*>*{background:radial-gradient(1q,#AA445F,106q,#E38F66);margin:80 130;border-radius:50%;border:32q solid;border-color:F7EC7D#0000;transform:rotate(45deg
```
now it' under 160 😄

**Reply by SelenIT** [reply]
There _is_ a shorter gradient solution... it just probably needs a bit more gradients :)

**Reply by gunnarbittersmann** [reply]
Iconic answer. 😆

**Reply by rafehimself** [reply]
@SelenIT what number do you mean ?cuz I have 161 with only gradients, but that still unsatisfying ...

**Reply by SelenIT** [reply]
I had 139 with gradients, recently learned that it can be optimized to 137

**Reply by verberg** [reply]
Thank's vor the tipp with the gradients. I tried and have now 155 chars. And it must give a better way!

**Comment by SelenIT** [comment, marked as answer]
When you set background for `*`, it means it applies to _all elements,_ in this case,`html` and `body`. This is why you get two "layers" of background. If you set background for `* > *`, it gets applied to `body` only, so you get only one `background` layer. However, here an interesting exception in the CSS standard comes into play: if the `body` element has a background and the `html` element doesn't, this background [is "propagated" to the `html` element](https://drafts.csswg.org/css-backgrounds/#body-background) and applied to it rather than to `body`. That's why the background of `body` is actually transparent if you set the background for `* > *`.

However, this doesn't explain why the background of the `body` element in 2-layer version (with `*`) has these areas covered with the background _color_ rather than _the gradient._ Apparently these areas "shine through" the semi-transparent pixels along the edge of the element (occured due to anti-aliasing), and cause this 0.1% of mismatch.
<img width="400" alt="Screenshot 2021-12-30 at 23 14 02" src="https://user-images.githubusercontent.com/282754/147785353-5078abbb-ad08-4a43-846e-ca77cc651bfd.png">

Even more interestingly, these areas disappear if you add a bit of transparency to the yellow parts of the border (e.g. `#F7EC7DFE` instead of `#F7EC7D`).

My guess is that it's a bug (or, at least, _peculiarity_) in the Chrome rendering engine: it assumes that the background area under opaque borders will be not visible, and doesn't paint the image (gradient) in that area, leaving only solid color there (probably it is meant to be kind of optimization). It "forgets" that parts of that area could be visible due to rounded corners, and just doesn't take anti-aliasing artifacts into account.

The workaround is to set only the gradient for the background, without adding solid color, as suggested by @rafehimself and as your working solution actually does.

**Reply by verberg** [reply]
Thank you for this very helpful explanation! 
And yes I found out, that it gets 100% when I have only one layer.
It also works, when I replace * width html.
But only in Chrome we can "see" the solution, in other browsers, it seems to be wrong, but gets 100%. It's crazy.

**Reply by SelenIT** [reply]
The backend of the CSSBattle uses Chrome engine to evaluate solutions, so rendering in other browsers isn't much relevant here (unfortunately or fortunately, depending on whether you aim for standard-compliant code or prioritize the code-golfing part, using all means, even hacks, to get the shortest solution possible 😊)

**Comment by gunnarbittersmann** [comment]
Microöptimization: Y’all know that the parser accepts `#RRGGBB` and `#RGB` values for the `color` property without the leading `#`?
This doesn’t work for the compound `background` property, but it would for `background-color` (which you would never use) **etc.**
